lib/ansible/modules/cloud/vmware/vmware_host_powerstate.py

```diff
@@ -0,0 +1,222 @@
+    - 'Not applicable if C(state) set to C(power-up-from-standby).'
+    choices: [ True, False ]
+    type: bool
+    default: False
+  timeout:
+    description:
+    - 'This parameter defines timeout for C(state) set to C(power-down-to-standby) or C(power-up-from-standby).'
+    - 'Ignored if C(state) set to C(reboot-host) or C(shutdown-host).'
+    - 'This parameter is defined in seconds.'
+    default: 600
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Set the state of a host system to reboot
+  vmware_host_powerstate:
+    hostname: 192.0.2.44
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: no
+    esxi_hostname: esxi01
+    state: reboot-host
+  delegate_to: localhost
+  register: reboot_host
+
+- name: Set the state of a host system to power down to standby
+  vmware_host_powerstate:
+    hostname: 192.0.2.44
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: no
+    esxi_hostname: power-down-to-standby
+    state: power-down-to-standby
+  delegate_to: localhost
+  register: power_down
+
+- name: Set the state of all host systems from cluster to reboot
+  vmware_host_powerstate:
+    hostname: 192.0.2.44
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: no
+    cluster_name: DC0_C0
+    state: reboot-host
+  delegate_to: localhost
+  register: reboot_host
+
+'''
+
+RETURN = r'''
+result:
+    description: metadata about host system's state
+    returned: always
+    type: dict
+    sample: {
+        "esxi01": {
+            "msg": "power down 'esxi01' to standby",
+            "error": "",
+        },
+    }
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task, TaskError
+from ansible.module_utils._text import to_native
+
+
+class VmwareHostPowerManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareHostPowerManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """
+        Function to manage internal state of host system
+
+        """
+        results = dict(changed=False, result=dict())
+        state = self.params.get('state')
+        force = self.params.get('force')
+        timeout = self.params.get('timeout')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            if not host.runtime.inMaintenanceMode and not force:
```

Comment:
Does this work if the host is powered down without being in maintenance mode (for instance in standby forced through this module) and you try to power it up from standby?

Reply:
Need to try.

Reply:
I am afraid that my setup does not provide facility to set host in standby mode.